istioctl/pkg/auth/upgrader.go

```diff
@@ -69,11 +75,39 @@ const (
 	requestAuthPrincipal = "request.auth.principal"
 )
 
-func NewUpgrader(k8sClient *kubernetes.Clientset, v1PolicyFiles, serviceFiles []string) *Upgrader {
+const (
+	rbacGlobalAllowAll = `apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: allow-all
+  namespace: {{ .RootNamespace }}
+spec:
+  rules:
+    - {}
+---
+`
+	rbacGlobalDenyAll = `apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: deny-all
+  namespace: {{ .RootNamespace }}
+spec:
+---
+`
+)
+
+const (
+	istioNamespace     = "istio-system"
+	istioConfigMapName = "istio"
```

Comment:
@yangminzhu I'm actually concerned about this. These values (configmap name and its namespace) can be different in the new installer, though I don't know if there are users who used the new installer to install Istio and have RBAC v1 policy.

Reply:
The alternative is providing a way for users to provide the configmap name and namespace, which I don't think it's much better than providing a way for them to just enter the root namespace. Wdyt?

Reply:
For the `istioNamespace`, I think we should do the same thing like the kube-inject command, which is adding a new flag called `istioNamespace` with default value `controller.IstioNamespace`.

For `meshConfigMapName`, we can also provide an extra option `meshConfigMapName` just like the kube-inject command but with the default value "istio".

For adding a flag for root namespace, that would be the easy implementation but I think it's complicating the user experience as it requires user to pass an extra flag even for the default installation and in order to provide the correct value, the user have to look up the root namespace in their config map manually and copy it here.

Reply:
No it's not about implementation. In this case instead of passing one flag (the root namespace), they have to pass in two flags (istioNamespace and meshConfigMapName), which they also  sometimes need to manually look them up. 

The differences between the new installer and the older installer plays a role here. In the old installer (which is what is used today to install Istio), the rootNamespace is set it to the releaseNamespace (default istio-system). The rootNamespace is located in the ConfigMap  named "istio" and lives in the releaseNamespace. So we can set the default behavior to get the rootNamespace directly using the API. If we couldn't do such thing, we can ask the users to manually look up the rootNamespace. I don't think it's any worse than manually looking up where the mesh config is, but instead of using 2 flags, the users only need one flag. Another thing is in the new installer, the rootNamespace is set to "istio-system" and I don't think it's configurable yet via the ConfigMap, so even if the users provide configmap name and its namespace, it won't work.

Reply:
So I feel like setting the default behavior to read from "istio" configmap in "istio-system" using the API. If not possible, manually provide the rootNamespace.

Reply:
If the default behavior is to get the configmap from the API server, then we always need a way to pass the value of `istioNamespace` and `istioConfigMapName` in case the default value doesn't work, that's why we need the extra flag to provide such values.

In the case the value is changed from the default, the user has the same problem when they are using the `kube-inject` command, we're not adding any additional burden here. 

I will be surprised if the new installer changes the default values here, that means it will immediately break the `istioctl kube-inject` command that is used everywhere on istio.io (e.g. https://istio.io/docs/examples/bookinfo/#start-the-application-services).

If the user changes these default values manually, it's easy for them to pass the new values to `istioctl kube-inject` and `istioctl x auth upgrade` in the exact same way, they don't need to learn anything new.

Reply:
> If the default behavior is to get the configmap from the API server, then we always need a way to pass the value of istioNamespace and istioConfigMapName in case the default value doesn't work, that's why we need the extra flag to provide such values.

Current it won't work with the new installer, because the rootNamespace is not in the MeshConfig (though we can update the configmap).

> I will be surprised if the new installer changes the default values here, that means it will immediately break the istioctl kube-inject command that is used everywhere on istio.io (e.g. https://istio.io/docs/examples/bookinfo/#start-the-application-services).

The default istioRootnamespace value is "istio-system" from both old and new installer. The new installer didn't change the default value.

Reply:
> Current it won't work with the new installer, because the rootNamespace is not in the MeshConfig (though we can update the configmap).

Then how does the new installer work with pilot? My understanding is that Pilot reads the rootNamespace from the MeshConfig which its contents comes from the configmap.

> The default istioRootnamespace value is "istio-system" from both old and new installer. The new installer didn't change the default value.

Then I don't see what's the point in your previous comment in https://github.com/istio/istio/pull/17897#discussion_r336588446, I thought the only concern is if the new installer uses a different default value in the configmap so that user need always to pass the extra flags to istioctl.

Reply:
Discussed offline. I updated the PR to make istio namespace and the config map name configurable.